navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -0,0 +1,260 @@
+    setContentView(R.layout.navigation_view_layout);
+    unpackBundle = savedInstanceState == null;
+    bind();
+
+    initMap(savedInstanceState);
+    initNavigation();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    if (locationLayer != null) {
+      locationLayer.onStop();
+    }
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+    if (locationLayer != null) {
+      locationLayer.onStop();
+    }
+  }
+
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+    if (navigation != null) {
+      navigation.onDestroy();
+    }
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    map = mapboxMap;
+    initLocation();
+    initLocationLayer();
+    initRoute();
+    initCamera();
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  @Override
+  public void onConnected() {
+    locationEngine.requestLocationUpdates();
+  }
+
+  @Override
+  public void onLocationChanged(Location location) {
+    this.location = location;
+    if (unpackBundle) {
+      unpackBundle(getIntent());
+      unpackBundle = false;
+    }
+  }
+
+  @Override
+  public void onProgressChange(Location location, RouteProgress routeProgress) {
+    locationLayer.forceLocationUpdate(location);
+    instructionView.update(routeProgress);
+  }
+
+  @Override
+  public void userOffRoute(Location location) {
+    Position newOrigin = Position.fromLngLat(location.getLongitude(), location.getLatitude());
+    fetchRoute(newOrigin, destination);
+  }
+
+  @Override
+  public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
+    if (validRouteResponse(response)) {
+      startNavigation(response.body().getRoutes().get(0));
+    }
+  }
+
+  @Override
+  public void onFailure(Call<DirectionsResponse> call, Throwable t) {
+    // TODO Send alertview message
+  }
+
+  private void bind() {
+    mapView = (MapView) findViewById(R.id.mapView);
+    instructionView = (InstructionView) findViewById(R.id.instructionView);
+  }
+
+  private void initMap(Bundle savedInstanceState) {
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  private void initNavigation() {
+    navigation = new MapboxNavigation(this);
+    navigation.addProgressChangeListener(this);
+
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  private void initLocation() {
+    //    locationEngine = new LocationSource(this);
```

Comment:
Why is this commented out?

Reply:
Was commented out while using the MockEngine to test - route simulation logic was added in https://github.com/mapbox/mapbox-navigation-android/pull/198/commits/b2a28e10acdaf030dacce69a058d2f89876379c8